src/main.cpp

```diff
@@ -85,10 +85,15 @@ static Result<void, TestError> run_program(InterpreterT& interpreter, ScriptOrMo
                 return visitor->parse_node();
             });
 
-        auto unit = JS::Bytecode::Generator::generate(program_node);
-        auto& passes = JS::Bytecode::Interpreter::optimization_pipeline();
-        passes.perform(*unit);
-        result = interpreter.run(*unit);
+        auto unit_result = JS::Bytecode::Generator::generate(program_node);
+        if (unit_result.is_error()) {
+            result = JS::throw_completion(JS::InternalError::create(interpreter.global_object(), unit_result.error().to_string()));
```

Comment:
Can this be `ErrorType::NotImplemented` (Or does the string already start with `TODO(`)

Since then it should track all such cases as TODO in the graphs and such.

Reply:
Oh yeah, that's much nicer to look at:
```
[ ✅ 18281 ❌ 8798  ⚙️ 112   💀 7     💥️ 7059  📝 11352 ]
```